### DISCUSSION TITLE
Make segment update local to the window, not global across all running tmux windows.

### DISCUSSION THREAD

**Discussion by tmjbios** [erikw/tmux-powerline#292] [Q&A] [answered]
Hi everyone.

My use case is 3-6 or more open gnome-terminals, all connecting to a single remote machine via ssh over a spotty VPN connection. I'm trying to use tmux-powerline to alleviate losing all my in-progress work when the VPN goes down.

However, the updates tmux-powerline puts out for the segments across the bottom are global; the pwd information for the current active gnome-terminal is reflected in all other windows running tmux. Same for the git branch, etc. Hit <enter> in a window 1, and all the other window's segments update to reflect the state of window 1 instead of keeping their data so I can track what window is in which dir, etc.

Is there a quick way to force the updates to be local to a session instead of global across all tmux processes? I've searched the 6+ pages of issues, and I'm not seeing anything describing what I see.

I usually start tmux with just "tmux" when I log in, but starting new sessions with -s doesn't help either. Am I missing something?

Thanks.

**Comment by erikw** [comment]
Hello @tmjbios,

could you tell me more about your connections. See the image [here](https://arcolinux.com/everthing-you-need-to-know-about-tmux-servers-and-sessions/) for terminology I refer to:

are each terminal window, via ssh, to the same server, using
* the same tmux server, same session (viewing same or different windows/panes?)
* the same tmux server, different sessions (same or different panes?)
* different tmux servers

Running locally one tmux server and two different sessions, the PWD segment seems to show the right directory although sometimes flimmer a bit showing some other directory.

![Screen Shot 2022-10-15 at 18 56](https://user-images.githubusercontent.com/175286/195998846-b98dbbd3-8d4a-430f-b9de-e229f5268fc5.png)


Sorry I don't know how to improve this. The relevant code is:
https://github.com/erikw/tmux-powerline/blob/887ed21108ec34614f2eaf4a93009c6cc4117ca2/lib/tmux_adapter.sh#L17-L19

**Comment by tmjbios** [comment, marked as answer]
One server, two sessions.

Running `tmux list-sessions` while I have two windows open, I see two sessions, 0 and 1, each with 1 window.

I'll poke around a bit and see if I can see what the get_tmux_cwd() is doing. Thanks for the reply!
